TRACING-4621: add troubleshooting guide for Instrumentation (#82280)
openshift/openshift-docs: modules/otel-troubleshooting-telemetry-data-generation-by-the-instrumentation-libraries.adoc

// Module included in the following assemblies:
//
// * observability/otel/otel-troubleshooting.adoc

:_mod-docs-content-type: PROCEDURE
[id="otel-troubleshooting-telemetry-data-generation-by-the-instrumentation-libraries_{context}"]
= Troubleshooting telemetry data generation by the instrumentation libraries

> Troubleshooting telemetry data generation by the instrumentation libraries involves checking the endpoint, looking for errors in your application logs, and verifying that the Collector is receiving the telemetry data.

Reviewer: ```suggestion
You can troubleshoot telemetry data generated by the instrumentation libraries by checking the endpoint, looking for errors in your application logs, and verifying that the Collector is receiving the telemetry data.
```

.Procedure

. Verify that the instrumentation is transmitting data to the correct endpoint:
+
[source,console]
----
oc get instrumentation <instrumentation_name> -n <workload_project> -o jsonpath='{.spec.endpoint}'
----
+

> The default endpoint `+http://localhost:4317+` for the `Instrumentation` object is only applicable to a Collector instance that is deployed as a sidecar in your application pod. So if you find that you are using an incorrect endpoint, you can correct it by editing the `Instrumentation` object and redeploying your application.

Reviewer: ```suggestion
The default endpoint `+http://localhost:4317+` for the `Instrumentation` object is only applicable to a Collector instance that is deployed as a sidecar in your application pod. If you are using an incorrect endpoint, correct it by editing the `Instrumentation` object and redeploying your application.
```

> . Inspect your application logs for error messages that might indicate that the instrumentation is malfunctioning:

Reviewer: ```suggestion
. Inspect your application logs for error messages that might indicate instrumentation malfunctions:
```

+
[source,console]
----
oc logs <application_pod> -n <workload_project>
----

> . If the application logs contain error messages that indicate that the instrumentation might be malfunctioning, you can install the OpenTelemetry SDK and libraries locally and run your application locally. This way, you can troubleshoot for issues between the instrumentation libraries and your application without {product}.

Reviewer 2: There is still a `{product}` variable in here that is rendering verbatim. Should it be `{product-title}`?

Also, see Janelle's suggestion from earlier at https://github.com/openshift/openshift-docs/pull/82280/files#r1837203741, which omits the `{product}` in the first place, if you like.

Reviewer 3: Thank you. I'm replacing `{product}` with `{product-title}` in my review PR.

> . If the application logs contain error messages that indicate that the instrumentation might be malfunctioning, you can install the OpenTelemetry SDK and libraries locally and run your application locally. This way, you can troubleshoot for issues between the instrumentation libraries and your application without {product}.

Reviewer: ```suggestion
. If the application logs' error messages indicate instrumentation malfunction, install the OpenTelemetry SDK and libraries locally and run your application locally.
```

> . Use the Debug Exporter to verify that the telemetry data is reaching the destination OpenTelemetry Collector instance. For more information, see the section about the Debug Exporter.

Reviewer 2: ```suggestion
. Use the Debug Exporter to verify that the telemetry data is reaching the destination OpenTelemetry Collector instance. For more information, see "Debug Exporter".
```

Reviewer 3: Thank you. Suggestion applied.

> . Use the Debug Exporter to verify that the telemetry data are reaching the destination OpenTelemetry Collector instance. For more information, see the section about the Debug Exporter.

Reviewer: ```suggestion
. Use the Debug Exporter to verify that the telemetry data is reaching the OpenTelemetry Collector instance. For more information, see the Debug Exporter.
```
